packages/flutter_tools/lib/src/android/android_emulator.dart

```diff
@@ -50,18 +54,59 @@ class AndroidEmulator extends Emulator {
 
   @override
   Future<void> launch() async {
-    final Future<void> launchResult = processUtils.run(
-      <String>[getEmulatorPath(), '-avd', id],
-      throwOnError: true,
+    final Process process = await processUtils.start(
+      <String>[getEmulatorPath(androidSdk), '-avd', id],
     );
-    // The emulator continues running on a successful launch, so if it hasn't
-    // quit within 3 seconds we assume that's a success and just return. This
-    // means that on a slow machine, a failure that takes more than three
-    // seconds won't be recognized as such... :-/
-    return Future.any<void>(<Future<void>>[
-      launchResult,
-      Future<void>.delayed(const Duration(seconds: 3)),
+
+    // Record output from the emulator process.
+    final List<String> stdoutList = <String>[];
+    final List<String> stderrList = <String>[];
+    final StreamSubscription<String> stdoutSubscription = process.stdout
+      .transform<String>(utf8.decoder)
+      .transform<String>(const LineSplitter())
+      .listen((String line) {
```

Comment:
Could be `.listen(stdoutList.add)`, here and below

Reply:
Good call. Done.